sdk/iot/common/inc/az_iot_common_internal.h

```diff
@@ -32,6 +32,16 @@
  */
 AZ_NODISCARD int32_t _az_iot_u32toa_size(uint32_t number);
 
+/**
+ * @brief Copies the url-encoded content of `source` span into `destination`, returning the free remaining of `destination`.
+ *
+ * @param[in] destination The span where the url-encoded content is copied to.
+ * @param[in] source The span to url-encode and copy the content from.
+ * @param[out] out_remainder A slice of `destination` with the non-used portion of `destination`.
+ * @return An `az_result` value.
+ */
+AZ_NODISCARD az_result _az_span_copy_url_encode(az_span destination, az_span source, az_span* out_remainder);
```

Comment:
_az_iot_span_url_encode ?

Reply:
This one is different. It url encodes, copies and **slices** the result to give the remaining buffer.